pkg/ingester/wal.go

```diff
@@ -0,0 +1,586 @@
+package ingester
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/go-kit/kit/log/level"
+	"github.com/golang/protobuf/proto"
+	"github.com/pkg/errors"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/prometheus/pkg/labels"
+	tsdb_errors "github.com/prometheus/tsdb/errors"
```

Comment:
These import from the old location, we should probably get them from prometheus/prometheus.

Reply:
I tried that, but I guess we need have another branch with tsdb in it because of the alertmanager problem we faced earlier. Once v2.12.0 is out, I will move to tsdb in prometheus by following the same process.